include/knowhere/dataset.h

```diff
@@ -71,18 +72,46 @@ class DataSet : public std::enable_shared_from_this<const DataSet> {
         this->data_[meta::DISTANCE] = Var(std::in_place_index<0>, dis);
     }
 
+    void
+    SetDistance(std::unique_ptr<float[]>&& dis) {
+        std::unique_lock lock(mutex_);
+        this->data_[meta::DISTANCE] = Var(std::in_place_index<0>, dis.release());
+    }
+
     void
     SetLims(const size_t* lims) {
         std::unique_lock lock(mutex_);
         this->data_[meta::LIMS] = Var(std::in_place_index<1>, lims);
     }
 
+    void
+    SetLims(std::unique_ptr<size_t[]>&& lims) {
+        std::unique_lock lock(mutex_);
+        this->data_[meta::LIMS] = Var(std::in_place_index<1>, lims.release());
+    }
+
     void
     SetIds(const int64_t* ids) {
         std::unique_lock lock(mutex_);
         this->data_[meta::IDS] = Var(std::in_place_index<2>, ids);
     }
 
+    void
+    SetIds(std::unique_ptr<long int[]>&& ids) {
+        static_assert(sizeof(long int) == sizeof(int64_t));
+
+        std::unique_lock lock(mutex_);
+        this->data_[meta::IDS] = Var(std::in_place_index<2>, reinterpret_cast<int64_t*>(ids.release()));
+    }
+
+    void
+    SetIds(std::unique_ptr<long long int[]>&& ids) {
```

Comment:
We we don't simply accept a `std::unique_ptr<int64_t[]>&& ids` ?

Reply:
`int64_t` is different on linux and on macos, this was the root of the problem why I implemented it this way